src/core/installAddon.js

```diff
@@ -151,6 +152,8 @@ export function makeProgressHandler({
       });
     } else if (event.type === 'onInstallFailed') {
       dispatch(setInstallError({ guid, error: INSTALL_FAILED }));
+    } else if (event.type === 'onCorruptFile') {
```

Comment:
Unfortunately, I discovered that FF does not expose this piece of information as a new type. The type is still `onDownloadFailed` but the `event.target.error` contains `ERROR_CORRUPT_FILE`, heh.